### PR TITLE
Add `--features unwind` to enable native extensions

### PR DESCRIPTION
**Pull request by pentschev**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by pentschev** [comment]
@conda-forge-admin, please rerender

**Comment by pentschev** [comment]
It looks like [ppc64](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=1073136&view=logs&j=76d26bd9-dbb3-560d-6d10-33f27642e45e&t=956f2f3c-0fa2-593c-d164-3da7ee40b74b) and [aarch64](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=1073136&view=logs&j=60238c3a-ba84-5ca5-39dc-717fbb421dbc&t=9cf66154-4ecc-5714-a3ee-b77d4233a250) builds fail during unwind build. @benfred do you happen to have any idea if those should work?

**Comment by pentschev** [comment]
After building and testing this locally I can confirm that with the changes from this PR `--native` is now available and functional on `x86_64`.

File: recipe/build.sh [hunk 1]
```diff
@@ -4,12 +4,14 @@ set -o xtrace -o nounset -o pipefail -o errexit
 
 if [ $(uname) = Darwin ] ; then
   export RUSTFLAGS="-C link-args=-Wl,-rpath,${PREFIX}/lib"
+  export BUILD_ARGS=""
 else
   export RUSTFLAGS="-C link-arg=-Wl,-rpath-link,${PREFIX}/lib -L${PREFIX}/lib"
+  export BUILD_ARGS="--features unwind"
```
Comment (benfred) [inline recipe/build.sh:19]: Right now this feature is only supported on x86_64 on linux - and I think is causing the PPC/aarch64 builds to fail. 

Maybe we should do something like this ?
```suggestion
  if [ $(dpkg --print-architecture) = amd64 ] ; then
    export BUILD_ARGS="--features unwind"
  else
    export BUILD_ARGS=""
  fi    
```

(fwiw, there has been some work started to enable this feature for aarch64 like in https://github.com/benfred/remoteprocess/pull/90/files - and I'd like to see the corresponding changes make it in to py-spy)

**Review by benfred — COMMENTED** [review]
thanks for the PR!

File: recipe/build.sh [hunk 2]
```diff
@@ -4,12 +4,14 @@ set -o xtrace -o nounset -o pipefail -o errexit
...
 fi
```
Comment (xhochy) [inline recipe/build.sh]: This should be the clean conda form:
```suggestion
if [[ "${target_platform}" == osx-* ]]; then
  export RUSTFLAGS="-C link-args=-Wl,-rpath,${PREFIX}/lib"
else
  export RUSTFLAGS="-C link-arg=-Wl,-rpath-link,${PREFIX}/lib -L${PREFIX}/lib"
fi

if [[ "${target_platform}" == "linux-64" ]]; then
  export BUILD_ARGS="--features unwind"
fi
```

File: recipe/build.sh (hunk 1 above)
Comment (xhochy) [reply recipe/build.sh:19]: See https://github.com/conda-forge/py-spy-feedstock/pull/36/files#r1831123083 for a suggestion on how to correctly check for the architecture.

Comment (pentschev) [reply recipe/build.sh:19]: Thanks @benfred for confirming and @xhochy for the suggestion.

@benfred should we also enable `--features unwind` for Windows or only Linux is supported?

Comment (benfred) [reply recipe/build.sh:19]: Yes - we support`--features unwind` on windows too

File: recipe/build.sh [hunk 3]
```diff
@@ -2,14 +2,18 @@
...
-if [ $(uname) = Darwin ] ; then
+if [[ "${target_platform}" == osx-* ]]; then
   export RUSTFLAGS="-C link-args=-Wl,-rpath,${PREFIX}/lib"
 else
   export RUSTFLAGS="-C link-arg=-Wl,-rpath-link,${PREFIX}/lib -L${PREFIX}/lib"
 fi
 
+if [[ "${target_platform}" == "linux-64" ]]; then
+  export BUILD_ARGS="--features unwind"
+fi
```
Comment (benfred) [inline recipe/build.sh:22]: Think we need to export BUILD_ARGS for non x86_64 linux here: 

```suggestion
else
  export BUILD_ARGS=""
fi  
```

**Comment by pentschev** [comment]
@conda-forge-admin, please rerender

**Comment by pentschev** [comment]
I think the builds for Linux and Windows should be correct, I do not have access to Windows machines to test this though.

@benfred @xhochy could you have a look whether anything else is needed here?

**Comment by xhochy** [comment]
I took the chance and updated the Rust compilation to what we use in other recipes nowadays.

**Comment by pentschev** [comment]
The changes look fine to me, although I do not have the knowledge to judge on their correctness, tests are also passing. I do not have any further comments, so from my end this is good to go if everyone else is satisfied.

Thanks @benfred @xhochy !

**Review by benfred — APPROVED** [review]
changes lgtm!

**Pull request merged**